tornado_sqlalchemy/__init__.py

```diff
@@ -148,18 +126,180 @@ def _make_session(self) -> Session:
 set_max_workers = _async_exec.set_max_workers
 
 
-def make_session_factory(*args, **kwargs):
-    return SessionFactory(*args, **kwargs)
+class SessionEx(Session):
+    """The SessionEx extends the default session system with bind selection.
+    """
+
+    def __init__(self, db, autocommit=False, autoflush=True, **options):
+        self.app = db.get_app()
+        bind = options.pop('bind', None) or db.engine
+        binds = options.pop('binds', db.get_binds())
+
+        Session.__init__(
+            self, autocommit=autocommit, autoflush=autoflush,
+            bind=bind, binds=binds, **options
+        )
+
+    def get_bind(self, mapper=None, clause=None):
+        """Return the engine or connection for a given model or
+        table, using the ``__bind_key__`` if it is set.
+        """
+        # mapper is None if someone tries to just get a connection
+        if mapper is not None:
+            try:
+                # SA >= 1.3
+                persist_selectable = mapper.persist_selectable
+            except AttributeError:
+                # SA < 1.3
+                persist_selectable = mapper.mapped_table
+
+            info = getattr(persist_selectable, 'info', {})
+            bind_key = info.get('bind_key')
+            if bind_key is not None:
+                return self.app.db.get_engine(bind=bind_key)
+        return Session.get_bind(self, mapper, clause)
+
+
+class BindMeta(DeclarativeMeta):
+    def __init__(cls, name, bases, d):
+        bind_key = (
+            d.pop('__bind_key__', None)
+            or getattr(cls, '__bind_key__', None)
+        )
+
+        super(BindMeta, cls).__init__(name, bases, d)
+
+        if bind_key is not None and getattr(cls, '__table__', None) is not None:
+            cls.__table__.info['bind_key'] = bind_key
+
+
+class SQLAlchemy(object):
```

Comment:
I think you can remove the `object` here because we're only supporting Python 3.